### DISCUSSION TITLE
Error: key with address696097B851A0ED2E353B9BEC022D3142DA124E5Dnot found: key not found

### DISCUSSION THREAD

**Discussion by perfogic** [cosmos/community#45] [Chain Builders (Cosmos SDK)]
Hello, I'm learning cosmos on this section https://tutorials.cosmos.network/hands-on-exercise/1-ignite-cli/4-create-message.html but I face this error any one can help me ?
**Here is my command:**
cosmos-checkersd keys add alice --keyring-backend test
cosmos-checkersd keys add bob --keyring-backend test

export alice=$(cosmos-checkersd keys show alice -a)
export bob=$(cosmos-checkersd keys show bob -a)

cosmos-checkersd tx cosmoscheckers create-game 

cosmos-checkersd tx cosmoscheckers create-game $alice $bob --from $alice --dry-run

**But I got this error:**
Error: key with address696097B851A0ED2E353B9BEC022D3142DA124E5Dnot found: key not found

**Debug:**
**Result Command to check keys list:**
cosmos-checkersd keys list
name: alice
type: local
address: cosmos1d9sf0wz35rkjudfmn0kqytf3gtdpynjaz5e2ue
pubkey: '{"@type":"/cosmos.crypto.secp256k1.PubKey","key":"AoLPcykQOzcndr+tDEUewY/HEsggA/MgJp8HKkp7D8en"}'
mnemonic: ""
name: bob
type: local
address: cosmos1g7tysyd4a74g2mv32sztadcyzsfuf9vuumfcxm
pubkey: '{"@type":"/cosmos.crypto.secp256k1.PubKey","key":"AjcHRgRyeNVI/PQsMEx6A4DoAO7ydCN+FQdOSUlaSEJn"}'
mnemonic: ""
**Command to run:**
cosmos-checkersd tx cosmoscheckers create-game cosmos1d9sf0wz35rkjudfmn0kqytf3gtdpynjaz5e2ue cosmos1g7tysyd4a74g2mv32sztadcyzsfuf9vuumfcxm  --from cosmos1d9sf0wz35rkjudfmn0kqytf3gtdpynjaz5e2ue --dry-run. Github: https://github.com/perfogic/cosmos-checkers

**Comment by julienrbrt** [comment]
Hi, you still need to precise the `--keyring-backend test` argument when submitting a tx, or set it as default in your `client.toml`.

**Reply by perfogic** [reply]
I define it still is not worked, however in client.toml it is default btw @@. I add --chain-id cosmoscheckers it work for broastcasting transaction (emit --dry-run) but still have that error when do with --dry-run

**Reply by julienrbrt** [reply]
I do think there were issue with --dry-run that were resolved in v0.46.

Have you tried to use this it a newer version of the SDK. v0.45.x is actually not supported anymore.

**Reply by perfogic** [reply]
oh thank you. I'm using v0.45.4
